src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -70,6 +79,67 @@ void ResumptionDataProcessor::Restore(ApplicationSharedPtr application,
   register_callbacks_[application->app_id()] = callback;
 }
 
+bool ResumptionDataProcessor::HasDataToRestore(
+    const smart_objects::SmartObject& saved_app) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const bool has_data_to_restore =
+      !saved_app[strings::application_submenus].empty() ||
+      !saved_app[strings::application_commands].empty() ||
+      !saved_app[strings::application_choice_sets].empty();
+
+  LOG4CXX_DEBUG(logger_, "HasDataToRestore: " << has_data_to_restore);
+
+  return has_data_to_restore;
+}
+
+bool ResumptionDataProcessor::HasGlobalPropertiesToRestore(
+    const smart_objects::SmartObject& saved_app) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& global_properties =
+      saved_app[strings::application_global_properties];
+
+  const bool has_gl_props_to_restore =
+      !global_properties[strings::help_prompt].empty() ||
+      !global_properties[strings::keyboard_properties].empty() ||
+      !global_properties[strings::menu_icon].empty() ||
+      !global_properties[strings::menu_title].empty() ||
+      !global_properties[strings::timeout_prompt].empty() ||
+      !global_properties[strings::vr_help].empty() ||
+      !global_properties[strings::vr_help_title].empty();
+
+  LOG4CXX_DEBUG(logger_,
```

Comment:
@mked-luxoft Use normal sentences for logging. `Application has global properties to restore`